### DISCUSSION TITLE
Authorization

### DISCUSSION THREAD

**Discussion by saleweaver** [saleweaver/python-amazon-sp-api#246] [Q&A]
Hi,
I would like to use your library to create a basic POC to get started. I'm a bit confused with the authorization because of the different Information I got. The Amazon Support told me I don't need a developer ID to authorize for the sp-api but in the documentation I read that I need a developer ID to get the LWA credentials. Can someone please tell me what is true here?
Thanks

**Comment by alexbenisch** [comment]
Hi saleweaver,
thanks for the reply.
Maybe I am missing something or the support staff was misinformed.

This is what I got from Amazon Support:

> AmazonSep 06, 2021 05:04 AM
> Afternoon.
> 
> Thank you for completing the developer profile for your Amazon Selling Partner API (SP-API).
> 
> Your request for access to SP-API has been APPROVED. You now have access to the SP-API for all global Marketplace sites based on the roles you requested.
> 
> 
> Please note that the SP-API does not require the use of a developer ID. To get started, you will need to create an IAM role that you provide with permissions to call the SP API when you register your application. You need an Amazon Web Service (AWS) account because the SP API security model uses AWS credentials. If you are not already an AWS customer, you can create a free AWS account. For more information, see Free AWS account: https://aws.amazon.com/free/
> 
> For more instructions, see the SP API Developer Guide at: Registering Your Vendor API App.
> 
> In addition, all apps offered to sellers on Amazon must be listed in the Marketplace Appstore. You can submit a new appstore listing form or edit an existing listing in Developer Central at: https://sellercentral.amazon.com/sellingpartner/developerconsole.
> 
> We look forward to featuring your application in the Marketplace Appstore.
>

**Reply by tector** [reply]
I have a question on this:
I don't want to publish an app in the Marketplace Appstore...

I have an already working LWA app with credentials which i use for the Advertising API - now i want to reuse the same credentials for the Selling Partner API.

Is that LWA "app"(?) not enough?

I have read something about if i want to use a "private app" it is necessary to create an app, but leave it as a 'draft'...  But i need more details on this to understand how it works...

EDIT: this app thing is very confusing: if i create an 'app' i have only the option to  choose between Android and Mobile Web, but what i think i need is something like 'API service only'

**Reply by saleweaver** [reply]
No, the credentials for another service are not valid. The authorization and everything App related is explained in Detail in Amazon's repository, I couldn't explain it better than they already did.

**Reply by tector** [reply]
well.. i think they doesn't explain it very well or i am too stupid :-/

I can't see the option to create an app which is NOT aiming for android or mobile...  i just want to query the Selling Partner API from my localhost environment or (in future) from a server...

**Reply by saleweaver** [reply]
I think you are looking somewhere in Login With Amazon, which is completely unrelated to the Selling Partner API. The App is setup in sellercentral and there is no option for android/... All you have to do is to read one page of documentation, and it's completely explained. https://github.com/amzn/selling-partner-api-docs/blob/main/guides/en-US/developer-guide/SellingPartnerApiDeveloperGuide.md#registering-as-a-developer

**Reply by tector** [reply]
Okay, now i understand: i have looked at the completely wrong place -> developers.amazon.com

I am already registered as developer for LWA and Advertising API, so i overlooked the part for registering in the documentation because i was not aware that you need to register >>AGAIN<< for another Amazon API.
Also the documentation >>DOESN'T<< link to the Registration Portal (https://developer.amazonservices.com/) so i overlooked that easily!

Thanks for your efforts!

**Reply by saleweaver** [reply]
Yeah, that's Amazon. The error you are getting is also because of the missing auth then.
